controllers/sr_controller/sr_controller.py

```diff
@@ -7,25 +7,65 @@
 # Root directory of the SR webots simulator (equivalent to the root of the git repo)
 ROOT = Path(__file__).resolve().parent.parent.parent
 
-ROBOT_FILE = ROOT.parent / "robot.py"
+MODE_FILE = ROOT / "robot_mode.txt"
 
 EXAMPLE_CONTROLLER_FILE = ROOT / "controllers/example_controller/example_controller.py"
 
+ROBOT_IDS_TO_CORNERS = {
+    "291": 0,
+    "684": 1,
+    "1077": 2,
+    "1470": 3,
+}
 
-if __name__ == "__main__":
-    if not ROBOT_FILE.exists():
+
+def get_robot_zone() -> int:
+    return ROBOT_IDS_TO_CORNERS[os.environ['WEBOTS_ROBOT_ID']]
+
+
+def get_robot_file() -> Path:
+    zone_id = get_robot_zone()
+    robot_file_dir = ROOT.parent / ("zone-" + str(zone_id))
+    if robot_file_dir.is_dir():
+        return robot_file_dir / "robot.py"
+    if get_robot_mode() == "comp":
+        print("WARNING: Using default robot file location in competition mode.")
+    return ROOT.parent / "robot.py"
+
+
+def get_robot_mode() -> str:
+    if not MODE_FILE.exists():
+        return "dev"
+    return MODE_FILE.read_text().strip()
+
+
+def main():
+    robot_file = get_robot_file()
+    robot_mode = get_robot_mode()
+    robot_zone = get_robot_zone()
+
+    if robot_mode == "dev" and not robot_file.exists():
         print("Robot controller not found, copying example into place.")
-        copyfile(str(EXAMPLE_CONTROLLER_FILE), str(ROBOT_FILE))
+        copyfile(str(EXAMPLE_CONTROLLER_FILE), str(robot_file))
+    elif not robot_file.exists():
+        print("ERROR: No robot controller found for zone ", robot_zone)
+        sys.exit(1)
```

Comment:
Just a thought -- would this be clearer if we hoisted the `not robot_file.exists()`?
``` python
    if not robot_file.exists():
        if robot_mode == "dev":
            print("Robot controller not found, copying example into place.")
            copyfile(str(EXAMPLE_CONTROLLER_FILE), str(robot_file))
        else:
            print("ERROR: No robot controller found for zone ", robot_zone)
            sys.exit(1)
```

Reply:
I was staring at this for ages trying to see how to make it look nicer, that's definitely it!